pants

```diff
@@ -2,6 +2,39 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P)
+
+# TODO(#7152): remove this after the python 3 migration!
+function clean_if_interpreter_changed() {
+  _python_version_to_run="$1"
+
+  if [[ -n "${_recursive_pants_shell_script_invocation:-}" ]]; then
+    return 0
+  fi
+
+  _python_version_file="${REPO_ROOT}/.pants-python-version"
+
+  if [[ -f "$_python_version_file" ]]; then
+    _prev_run_python_version="$(cat "$_python_version_file")"
+    if [[ "$_python_version_to_run" == "$_prev_run_python_version" ]]; then
+      return 0
+    fi
+  else
+    cat >&2 <<EOF
+${_python_version_file} not found. Forcing an initial clean-all...
+EOF
+  fi
+
+  export _recursive_pants_shell_script_invocation='y'
+  cat >&2 <<EOF
+Caches must be cleared before switching pants Python versions in this repo.
+Preparing a Python ${_python_version_to_run} invocation...
```

Comment:
I love that this prints the full python version, including patch. Helpful to know what exactly I'm running, e.g. Pyenv Python 2.7.15 vs system 2.7.13!